Removing WebDriverMonkeyPatches methods

aaltat opened the pull request:
Removed get_current_url, get_page_source, get_title and
get_window_handles methods from WebDriverMonkeyPatches. Made the
library to the counterpart from the selenium side directly.
Added acceptance test for Get Title keyword and fixed the unit
test to use the selenium methods.

Todo: Remove the rest of the methods or implement methods somewhere else in the code base.

Commit:
Removing WebDriverMonkeyPatches methods

Removed get_current_url, get_page_source, get_title and
get_window_handles methods from WebDriverMonkeyPatches. Made the
library to the counterpart from the selenium side directly.
Added acceptance test for Get Title keyword and fixed the unit
test to use the selenium methods.

pekkaklarck commented:
On a quick look this looks good to be merged. Or do you want to remove other monkeys first?

emanlove commented:
Question, which I think @aaltat answered in his post under #706, is this backward compatible? Reading through the code it looks like it is but I want to verify that we are indeed still backward compatible.

emanlove commented on test/unit/test_webdrivermonkeypatches.py:
```diff
@@ -1,69 +0,0 @@
-    def test_window_info_values_are_strings(self):
-        driver = MockWebDriver()
-        when(driver).execute_script(SCRIPT).thenReturn(['id', 'name', 'title', 'url'])
-        info = driver.get_current_window_info()
```
I am be missing something here but it looks like all these unit tests just test get_current_window_info method which I don't see being removed above. I can see removing these unit tests if get_current_window_info is removed but as it is not yet is this premature to remove these unit tests?

aaltat replied:
True, will modify the file accordingly

aaltat replied:
Added file back.

aaltat replied:
Renamed the file, because removed corresponding monkey patch. Also made the unit test work with the new implementation.

aaltat committed:
Added test_webdrivermonkeypatches.py back

aaltat commented:
Are we happy and can this be merged?

pekkaklarck commented:
This doesn't remove all monkeys yet, right?

aaltat committed:
Removed get_current_window_handle() monkey patch

aaltat committed:
Removing get_current_window_info from webdrivermonkeypatches

The get_current_window_info method is removed from
webdrivermonkeypatches and added to windowmanager with name
_get_current_window_info.

aaltat committed:
Removed current_window_is_main from webdrivermonkeypatches

aaltat committed:
S2L set_selenium_speed and webdrivermonkeypatches re-factored

By default, S2L did always monkey patch the WebDriver execute method.
But the monkey patch was actually needed actually only needed when the
S2L set_selenium_speed method was called. Re-factored the code, to
perform the monkey patching only when the S2L set_selenium_speed
method is called.

aaltat committed:
Minor refactoring to make code base more cleaner.
